pkg/systemd/overview-cards/cryptoPolicies.jsx

```diff
@@ -76,41 +80,49 @@ export const CryptoPolicyRow = () => {
                 <CryptoPolicyDialog close={() => setIsOpen(false)}
                                     currentCryptoPolicy={currentCryptoPolicy}
                                     setCurrentCryptoPolicy={setCurrentCryptoPolicy}
+                                    fipsEnabled={fipsEnabled}
                 />
                 }
             </td>
         </tr>
     );
 };
 
+const setPolicy = (policy, setError, setInProgress) => {
+    setInProgress(true);
+
+    const promises = [];
+    if (policy === "FIPS") {
+        promises.push(cockpit.spawn(["fips-mode-setup", "--enable"], { superuser: "require", err: "message" }));
+    } else {
+        promises.push(cockpit.spawn(["fips-mode-setup", "--disable"], { superuser: "require", err: "message" }).then(() =>
+            cockpit.spawn(["update-crypto-policies", "--set", policy], { superuser: "require", err: "message" })));
+    }
```

Comment:
This is always exactly one promise, so turn this into a `const promise` with a ternary.

Or maybe even better, don't chain the f-m-s and u-c-p promises, but run them in parallel. I think u-c-p doesn't mind, right? Then you can also factorize the f-m-s call and just conditionalize the `--enable` vs. `--disable`, not the whole spawn.